yt/utilities/on_demand_imports.py

```diff
@@ -67,7 +76,7 @@ def Dataset(self):
         return self._Dataset
 
 
-_netCDF4 = netCDF4_imports()
+_netCDF4 = netCDF4_imports()  # Always do that before initializing h5py_imports() !!!
```

Comment:
Could the comment also include an explanation ?

Reply:
yes haha -- explained above in another comment, but this comment should be clearer here...